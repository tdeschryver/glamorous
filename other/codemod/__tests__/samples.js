const fs = require('fs')
const path = require('path')
const babel = require('babel-core')

test('test every sample!', async () => {
  const files = await getFiles('samples')

  expect.assertions(files.length)
  await Promise.all(
    files.map(async filename => {
      const stripped = path.basename(filename, path.extname(filename))
      const actual = await transformFileAsync('samples', stripped)
      expect(actual).toMatchSnapshot(filename)
    }),
  )
})

function getFiles(subfolder) {
  return new Promise((resolve, reject) => {
    fs.readdir(
      path.join(__dirname, `./fixtures/${subfolder}`),
      (err, result) => {
        if (err) {
          reject(err)
        }

        resolve(result)
      },
    )
  })
}

function transformFileAsync(subfolder, filename) {
  const opts = Object.assign(
    {
      extends: path.join(__dirname, '../.babelrc'),
    },
    {},
  ) //TODO: options

  return new Promise((resolve, reject) => {
    babel.transformFile(
      path.join(__dirname, `./fixtures/${subfolder}/${filename}.js`),
      opts,
      (err, {code} = {}) => {
        if (err) {
          reject(err)
        }

        resolve(code)
      },
    )
  })
}
