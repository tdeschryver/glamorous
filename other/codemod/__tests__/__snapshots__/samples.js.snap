// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eight.js 1`] = `
"const Eight = glamorous.div((
  {
    important,

    theme: {
      main: {fontSize: theFontSize}
    }
  }
) => ({
  fontSize: theFontSize,
  color: important ? 'red' : 'black'
}))
"
`;

exports[`five.js 1`] = `
"const Five = glamorous.div(props => ({
  fontSize: props.theme.main.fontSize,
  color: props.important ? 'red' : 'black'
}))
"
`;

exports[`four.js 1`] = `
"const Four = glamorous.div((
  {
    important,
    theme
  }
) => ({
  fontSize: theme.main.fontSize,
  color: important ? 'red' : 'black'
}))
"
`;

exports[`nine.js 1`] = `
"const Nine = glamorous.div(props => ({
  fontSize: props.theme.main.fontSize,
  fontWeight: props.theme.main.fontWeight,
  color: props.important ? 'red' : 'black'
}))
"
`;

exports[`one.js 1`] = `
"const One = glamorous.div((
  {
    theme
  }
) => ({
  fontSize: theme.main.fontSize
}))
"
`;

exports[`seven.js 1`] = `
"const Seven = glamorous.div(({theme}) => ({
  fontSize: theme.main.fontSize,
}))
"
`;

exports[`six.js 1`] = `
"const Six = glamorous.div((
  {
    important,

    theme: {
      main
    }
  }
) => ({
  fontSize: main.fontSize,
  color: important ? 'red' : 'black'
}))
"
`;

exports[`three.js 1`] = `
"const Three = glamorous.div((
  {
    important,
    theme
  }
) => ({
  fontSize: theme.main.fontSize,
  color: important ? 'red' : 'black'
}))
"
`;

exports[`two.js 1`] = `
"
const Two = glamorous.div((
  {
    theme
  },
  tripple
) => ({
  fontSize: theme.main.fontSize
}))
"
`;
